[Bug] flink failed to create tag during the day period, tagTotime parse error

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.


### Paimon version

0.5-SNAPSHOT

### Compute Engine

Flink-1.16.1

### Minimal reproduce step

no

### What doesn't meet your expectations?

switched from INITIALIZING to FAILED with failure cause: java.time.format.DateTimeParseException: Text '2023-08-08' could not be parsed: Unable to obtain LocalDateTime from TemporalAccessor: {},ISO resolved to 2023-08-08 of type java.time.format.Parsed
	at java.time.format.DateTimeFormatter.createError(DateTimeFormatter.java:1920)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1855)
	at java.time.LocalDateTime.parse(LocalDateTime.java:492)
	at org.apache.paimon.tag.TagAutoCreation$BaseTagPeriodHandler.tagToTime(TagAutoCreation.java:225)
	at org.apache.paimon.tag.TagAutoCreation.<init>(TagAutoCreation.java:112)
	at org.apache.paimon.tag.TagAutoCreation.create(TagAutoCreation.java:337)
	at org.apache.paimon.AbstractFileStore.newTagCreationManager(AbstractFileStore.java:230)
	at org.apache.paimon.table.AbstractFileStoreTable.newCommit(AbstractFileStoreTable.java:263)
	at org.apache.paimon.flink.sink.FlinkWriteSink.lambda$createCommitterFactory$99e3de04$1(FlinkWriteSink.java:52)
	at org.apache.paimon.flink.sink.CommitterOperator.initializeState(CommitterOperator.java:108)
	at org.apache.flink.streaming.api.operators.StreamOperatorStateHandler.initializeOperatorState(StreamOperatorStateHandler.java:122)
	at org.apache.flink.streaming.api.operators.AbstractStreamOperator.initializeState(AbstractStreamOperator.java:283)
	at org.apache.flink.streaming.runtime.tasks.RegularOperatorChain.initializeStateAndOpenOperators(RegularOperatorChain.java:106)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.restoreGates(StreamTask.java:726)
	at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$1.call(StreamTaskActionExecutor.java:55)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.restoreInternal(StreamTask.java:702)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.restore(StreamTask.java:669)
	at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:935)
	at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:904)
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:728)
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:550)
	at java.lang.Thread.run(Thread.java:748)
	Suppressed: java.lang.NullPointerException
Caused by: java.time.DateTimeException: Unable to obtain LocalDateTime from TemporalAccessor: {},ISO resolved to 2023-08-08 of type java.time.format.Parsed
	at java.time.LocalDateTime.from(LocalDateTime.java:461)
	at java.time.format.Parsed.query(Parsed.java:226)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
	... 20 more
Caused by: java.time.DateTimeException: Unable to obtain LocalTime from TemporalAccessor: {},ISO resolved to 2023-08-08 of type java.time.format.Parsed
	at java.time.LocalTime.from(LocalTime.java:409)
	at java.time.LocalDateTime.from(LocalDateTime.java:457)
	... 22 more

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java
+++ b/paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java
@@ -274,8 +274,8 @@ public class TagAutoCreation {
         protected DateTimeFormatter formatter() {
             return DAY_FORMATTER;
         }
-		
-		@Override
+
+        @Override
         public LocalDateTime tagToTime(String tag) {
             return LocalDate.parse(tag, formatter()).atStartOfDay();
         }

--- a/paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java
+++ b/paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java
@@ -274,6 +274,11 @@ public class TagAutoCreation {
         protected DateTimeFormatter formatter() {
             return DAY_FORMATTER;
         }
+		
+		@Override
+        public LocalDateTime tagToTime(String tag) {
+            return LocalDate.parse(tag, formatter()).atStartOfDay();
+        }
     }
 
     private static class TwoHoursTagPeriodHandler extends BaseTagPeriodHandler {

--- a/paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java
+++ b/paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java
@@ -29,6 +29,7 @@ import javax.annotation.Nullable;
 
 import java.time.Duration;
 import java.time.Instant;
+import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;

--- a/paimon-core/src/test/java/org/apache/paimon/tag/TagAutoCreationTest.java
+++ b/paimon-core/src/test/java/org/apache/paimon/tag/TagAutoCreationTest.java
@@ -144,7 +144,7 @@ public class TagAutoCreationTest extends PrimaryKeyTableTestBase {
     }
 
     @Test
-    public void testTagDaily() {
+    public void testTagDaily() throws Exception {
         Options options = new Options();
         options.set(TAG_AUTOMATIC_CREATION, TagCreationMode.WATERMARK);
         options.set(TAG_CREATION_PERIOD, TagCreationPeriod.DAILY);
@@ -159,6 +159,13 @@ public class TagAutoCreationTest extends PrimaryKeyTableTestBase {
         // test second create
         commit.commit(new ManifestCommittable(0, utcMills("2023-07-19T12:00:01")));
         assertThat(tagManager.tags().values()).containsOnly("2023-07-17", "2023-07-18");
+
+        // test newCommit create
+        commit.close();
+        commit = table.newCommit(commitUser).ignoreEmptyCommit(false);
+        commit.commit(new ManifestCommittable(0, utcMills("2023-07-20T12:00:01")));
+        assertThat(tagManager.tags().values())
+                .containsOnly("2023-07-17", "2023-07-18", "2023-07-19");
     }
 
     private long utcMills(String timestamp) {

